### ISSUE TITLE
Synapse 1.50.0 returns 404 Unrecognized request for all client requests if a `webclient` resource is enabled

### ISSUE THREAD

**Issue by anoadragon453** [matrix-org/synapse#11763]
TL;DR if you have the `client` and `webclient` resources enabled on a port in your `listeners` config section, attempts to access `client` resources may fail with a `404`, **breaking client functionality**. Workaround is to either:

* Re-order your `resources` config option such that `webclient` appears before `client`. Example:
    ```yaml
    resources:
      - names: [webclient, client, ...]
      ...
   ```

    though note that if you rely on the `webclient` resource, this workaround will render it inaccessible. Thus it is functionally equivalent to the next workaround:

* Remove `webclient` from your list of resources, or
* [downgrade](https://matrix-org.github.io/synapse/latest/upgrade#rolling-back-to-older-versions) to Synapse 1.49.2.

First noticed in https://github.com/matrix-org/synapse/issues/11762.

This first appears to be related to changes made in https://github.com/matrix-org/synapse/pull/11602

**Comment by anoadragon453** [comment]
A fix for this issue was implemented in https://github.com/matrix-org/synapse/pull/11764 has been released in [Synapse v1.50.1](https://github.com/matrix-org/synapse/releases/tag/v1.50.1).
